### PR TITLE
warnings: squashing more warnings 

### LINKED ISSUE #3822: build/warnings: ICC warning #20 on #pragma weak
```
 warning.src/binding/fortran/mpif_h/register_datarepf.c:280
 Error Details

src/binding/fortran/mpif_h/register_datarepf.c(280): warning #20: identifier mpi_conversion_fn_null__ is undefined
```
The line it was complaining:
```
#pragma weak mpi_conversion_fn_null__ = mpi_conversion_fn_null_
#pragma weak mpi_conversion_fn_null   = mpi_conversion_fn_null_
#pragma weak MPI_CONVERSION_FN_NULL   = mpi_conversion_fn_null_
```
They are weak symbols, so they are expected to be undefined. I don't understand why `icc` warns about them.

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

On top of PR #3895, this PR tries to make more-warnings tests clean.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)

Fixes #3822

* [ ] Passes tests (included warning check)
* [ ] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [ ] Commits are self-contained and do not do two things at once
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/warnings/more

**Comment by hzhou** [comment]
test:mpich/warnings/more

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:more/warnings

**Comment by hzhou** [comment]
test:mpich/warnings/more

**Comment by hzhou** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

File: src/binding/fortran/mpif_h/buildiface [hunk 1]
```diff
@@ -5317,6 +5317,10 @@ sub HelperForRegister_datarep {
 /* Add the prototype so the routine knows what this is */
 extern FORT_DLL_SPEC int FORT_CALL mpi_conversion_fn_null_ ( void*v1, MPI_Fint*v2, MPI_Fint*v3, void*v4, MPI_Offset*v5, MPI_Fint *v6, MPI_Fint*v7, MPI_Fint *ierr );
 
+extern FORT_DLL_SPEC int FORT_CALL mpi_conversion_fn_null__ ( void*v1, MPI_Fint*v2, MPI_Fint*v3, void*v4, MPI_Offset*v5, MPI_Fint *v6, MPI_Fint*v7, MPI_Fint *ierr );
```
Comment (raffenet) [inline src/binding/fortran/mpif_h/buildiface]: whitespace nit: why a blank line between these and the first decl?

Comment (hzhou) [reply src/binding/fortran/mpif_h/buildiface]: I'll fix it after confirming the test the results.

**Comment by hzhou** [comment]
1 failure if `ofi-external`  over `rma/rget_testall 2`. Re-run to confirm:
test:jenkins/ch4/ofi

**Comment by hzhou** [comment]
The failure doesn't seem to repeat. It is very curious though. Pasting the log:
```
not ok 957 - ./rma/rget_testall 2
  ---
  Directory: ./rma
  File: rget_testall
  Num-procs: 2
  Timeout: 180
  Date: "Tue Aug 20 16:39:12 2019"
  ...
## Test output (expected 'No Errors'):
## 1 - Got val1 = 0, val2 = 0, expected 1, 2
##  Found 1 errors
```

**Comment by hajimefu** [comment]
@hzhou We've been seeing the same `rget_testall` failure from time to time. Just filed #4011 for future reference.

**Pull request merged**